packages/contracts/contracts/core/plugin/PluginTransparentUpgradeable.sol

```diff
@@ -0,0 +1,42 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.10;
+
+import "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+import "@openzeppelin/contracts-upgradeable/utils/introspection/ERC165Upgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/proxy/utils/UUPSUpgradeable.sol";
+
+import {PermissionManager} from "../permission/PermissionManager.sol";
+
+import {AppStorage} from "../../utils/AppStorage.sol";
+import {IDAO} from "../IDAO.sol";
+import {DaoAuthorizableUpgradeable} from "../component/DaoAuthorizableUpgradeable.sol";
+
+/// @title PluginTranparentUpgradable
+/// @notice An abstract contract to inherit from when creating a proxy contract.
+/// This should be used to deploy logic contracts where proxy itself
+/// is deployed through transparent or beacon...
+abstract contract PluginTransparentUpgradeable is
```

Comment:
Below we extend from ContextUpgradeable and AppStorage. Aren't they useful here?

Reply:
fixed to use context.
in this iteration we go with all plugins to use appStorage